### DISCUSSION TITLE
How to add fix installation steps in the Docker-server file?

### DISCUSSION THREAD

**Discussion by purohit2103** [ibm-messaging/mq-container#585] [General]
Hi,

I am trying to figure out how to add fix installation steps to the Docker-server file.

What is the ask: 

Install 9.3.5.0-IBM-MQ-LinuxX64-LAIT45974-317489.tar for MQ 9.3.5 version

What is not working:

###############################################################################
# Main build stage, to build MQ image
###############################################################################
FROM $BASE_IMAGE:$BASE_TAG AS mq-server
ARG MQ_URL
ARG BASE_IMAGE
ARG BASE_TAG
ARG GO_WORKDIR
LABEL summary="IBM MQ Advanced Server" \
  description="Simplify, accelerate and facilitate the reliable exchange of data with a security-rich messaging solution — trusted by the world’s most successful enterprises" \
  vendor="IBM" \
  maintainer="IBM" \
  distribution-scope="private" \
  authoritative-source-url="https://www.ibm.com/software/passportadvantage/" \
  url="https://www.ibm.com/products/mq/advanced" \
  io.openshift.tags="mq messaging" \
  io.k8s.display-name="IBM MQ Advanced Server" \
  io.k8s.description="Simplify, accelerate and facilitate the reliable exchange of data with a security-rich messaging solution — trusted by the world’s most successful enterprises" \
  base-image=$BASE_IMAGE \
  base-image-release=$BASE_TAG
COPY --from=mq-redux /opt/mqm-redux/ /opt/mqm/
**# Customization
# Adding the Rolling update fix tar to /opt/mqm dir, applying the fix and deleting the untarred dir
ADD 9.3.5.0-IBM-MQ-LinuxX64-LAIT45974-317489.tar.gz /opt/mqm
RUN cd /opt/mqm/9.3.5.0-IBM-MQ-LinuxX64-LAIT45974-317489 && ./mqfixinst.sh /opt/mqm install 9.3.5.0-IBM-MQ-LinuxX64-LAIT45974-317489 \
  && cd - && rm -rf /opt/mqm/9.3.5.0-IBM-MQ-LinuxX64-LAIT45974-317489
# End**

What is the issue:

mqfixinst.sh fails at steps where it is trying to change ownership of certain files to mqm:mqm. We have tried to replace mqm:mqm with 1001:root as thats what is being used in other sections of the Docker-server file. But now its failing as its unable to find the actual fix files under ship directory. So it looks like probably what we are trying is not the right approach.

Looking for some pointers or documentation on how to get this working. Please advise.

**Comment by arthurbarr** [comment]
The main thing is to make sure that you're in a section of the Dockerfile which has `root` permissions.  i.e. some of the Dockerfile runs under a section `USER 0` (root), and then later it changes to `USER 1001`.  You will need root permissions during the build to do certain things.

You're right that a traditional fix which assumes the `mqm` user won't work, and does indeed need to be `1001:0` as you say.

**Comment by purohit2103** [comment]
We had raised a PMR for this and as per the suggestion given there (mentioned below), we made the changes and it worked

"Could you please ask the user to add the same lines they mentioned in the above git issue before this line - https://github.com/ibm-messaging/mq-container/blob/9.3.5/Dockerfile-server#L147 and give it a try. It will work. "

Thanks for the assistance!
